Author: Vincent Bernat <bernat@debian.org>
Last-Updated: Andreas Tille <tille@debian.org>  Tue, 08 Mar 2016 11:15:04 +0100
Description: Ensure that the default keymap is US english if there is a problem locating one.

--- a/xrdp/lang.c
+++ b/xrdp/lang.c
@@ -258,7 +258,15 @@ get_keymaps(int keylayout, struct xrdp_k
     }
     else
     {
+      if (keylayout != 0x409)
+      {
+        g_free(filename);
+        log_message(LOG_LEVEL_WARNING, "keymap for 0x%4.4x was not found. Falling back to 0x0409 instead",
+                    keylayout);
+        return get_keymaps(0x409, keymap);
+      } else {
         log_message(LOG_LEVEL_WARNING, "File does not exist: %s", filename);
+      }
     }
 
     g_free(filename);
